src/components/shopping.js

```diff
@@ -1,9 +1,24 @@
-import React from 'react';
+import React, { useState } from 'react';
+import getToken from '../../src/tokenScript';
+import TestList from '../lib/testList';
 import '../App.css';
 
-class Shopping extends React.Component {
-  render() {
-    return <h1>Shopping List</h1>;
+function Shopping() {
+  const [user_token, refreshToken] = useState(
+    localStorage.getItem('user_token'),
```

Comment:
A few things to point out here:

I would save `user_token` as a constant. It's a pain tracking down a string typo for a key like this.

Another thing to do here that would optimize how we read from `localStorage` is to use a callback function:

```jsx
useState(() => localStorage.getItem('user_token'))
```

Doing so will ensure we only read `localStorage` on the first render for `useState` rather than _every_ render.

It's common practice to stick with a single naming convention, like camelCase all variable names (ie. `userToken`).

Reply:
We did create a constant for USER_TOKEN. And used camel case for `userToken`.

We also created a custom hook, but we're very uncertain as to if this is really doing us any good that we couldn't have gotten from `useState`.

Reply:
This logic is now in `App.js`.